packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts

```diff
@@ -121,8 +163,10 @@ export class ApplicationListenerRule extends cdk.Construct {
       throw new Error(`At least one of 'hostHeader' or 'pathPattern' is required when defining a load balancing rule.`);
     }
 
-    if (props.targetGroups && props.fixedResponse) {
-      throw new Error('Cannot combine `targetGroups` with `fixedResponse`.');
+    if ((props.targetGroups && props.fixedResponse) ||
```

Comment:
This way of checking is not going to scale when we add yet another type of response.

Please do something similar to what I requested here: https://github.com/aws/aws-cdk/pull/4622#discussion_r337453320